File: REPASO PRUEBA U3/PY/utils/mongo_connection.py
from pymongo import MongoClient

class MongoConnection:
    _instance = None
    _client = None
    _database = None

    # Mismo string de conexión
    CONNECTION_STRING = "mongodb+srv://Mathews:Mathews2007@cluster0.6l9ibfh.mongodb.net/?appName=Cluster0"
    DATABASE_NAME = "ArtGalleryDB"

    @classmethod
    def get_database(cls):
        if cls._database is None:
            try:
                cls._client = MongoClient(cls.CONNECTION_STRING)
                cls._database = cls._client[cls.DATABASE_NAME]
                print("Conexión a MongoDB exitosa.")
            except Exception as e:
                print(f"Error conectando a MongoDB: {e}")
        return cls._database

    @classmethod
    def close(cls):
        if cls._client:
            cls._client.close()
            print("Conexión a MongoDB cerrada.")

File: REPASO PRUEBA U3/RepasoPrueba/src/main/java/ec/edu/espe/repasoprueba/utils/MongoConnection.java
package ec.edu.espe.repasoprueba.utils;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;    
/**
 *
 * @author Mathews Pastor, The POOwer Rangers Of Programming
 */
public class MongoConnection {

    private static MongoConnection instance;
    private MongoClient mongoClient;
    private MongoDatabase database;

    private static final String CONNECTION_STRING = "mongodb+srv://Mathews:Mathews2007@cluster0.6l9ibfh.mongodb.net/?appName=Cluster0";
    private static final String DATABASE_NAME = "ArtGalleryDB";

    private MongoConnection() {
        try {
            // Crea la conexión al cliente de MongoDB
            mongoClient = MongoClients.create(CONNECTION_STRING);
            database = mongoClient.getDatabase(DATABASE_NAME);
            System.out.println("Conexión a MongoDB exitosa.");
        } catch (Exception e) {
            System.err.println("Error conectando a MongoDB: " + e.getMessage());
        }
    }

    // Método estático para obtener la única instancia
    public static MongoConnection getInstance() {
        if (instance == null) {
            instance = new MongoConnection();
        }
        return instance;
    }

    // Retorna la base de datos para realizar operaciones
    public MongoDatabase getDatabase() {
        return database;
    }

    // Cierra la conexión al finalizar la aplicación
    public void close() {
        if (mongoClient != null) {
            mongoClient.close();
            System.out.println("Conexión a MongoDB cerrada.");
        }
    }
}
